src/borg/compress.pyx

```diff
@@ -222,22 +243,36 @@ class Auto(CompressorBase):
     ID = None
     name = 'auto'
 
-    logger = create_logger('borg.debug.file-compression')
-
     def __init__(self, compressor):
         super().__init__()
         self.compressor = compressor
         self.lz4 = get_compressor('lz4')
         self.none = get_compressor('none')
 
-    def compress(self, data):
+    def _decide(self, data):
+        """
+        Decides what to do with *data*. Returns (compressor, lz4_data).
+
+        *lz4_data* is the LZ4 result if *compressor* is LZ4 as well, otherwise it is None.
+        """
         lz4_data = self.lz4.compress(data)
-        if len(lz4_data) < 0.97 * len(data):
-            return self.compressor.compress(data)
-        elif len(lz4_data) < len(data):
-            return lz4_data
+        ratio = len(lz4_data) / len(data)
```

Comment:
ok. I also couldn't find a case where this blows up, just the theoretical possibility.